bishnu-suyel/Druid-Final-Project
# Tdfp52 mautic

# In this PR

- Refactored Mautic_integration module to remove paragraph visibility manipulation and taxonomy term creation
- Added segmentService.js to handle the storage and updates of Mautic segments that are fetched from the Drupal
- Modified dataParser.js to handle paragraph filtering and subsequent rendering based on the match between paragraph field segment and Mautic segment

# To test

- import db and configs
- check in Mautic dashboard your contact segment, you should see the corresponding to your segment text in the Homepage (contact, visitor or no segment)
- you can also manually choose segment in Drupal UI in the Segment select list in each Paragraph OR you can change your contact segment in Mautic and see the content updates in the Homepage
- if you want to switch between Mautic segments for testing, run the following command in the console `sessionStorage.setItem("segmentTestMode", "true");`, this will enable segment data updates every 5 seconds. Run `sessionStorage.removeItem("segmentTestMode");` to revert to the default update interval of 60 min.
